src/main/scala/com/campudus/tableaux/controller/StructureController.scala

```diff
@@ -0,0 +1,97 @@
+package com.campudus.tableaux.controller
+
+import com.campudus.tableaux.ArgumentChecker._
+import com.campudus.tableaux.TableauxConfig
+import com.campudus.tableaux.database._
+import com.campudus.tableaux.database.domain._
+import com.campudus.tableaux.database.model.StructureModel
+import com.campudus.tableaux.database.model.TableauxModel._
+
+import scala.concurrent.Future
+
+object StructureController {
+  def apply(config: TableauxConfig, repository: StructureModel): StructureController = {
+    new StructureController(config, repository)
+  }
+}
+
+class StructureController(override val config: TableauxConfig, override protected val repository: StructureModel) extends Controller[StructureModel] {
+
+  lazy val tableStruc = repository.tableStruc
+  lazy val columnStruc = repository.columnStruc
```

Comment:
I don't think that we need `lazy` here as `repository` is already available and `tableStruc` and `columnStruc` are `val` in `StructureModel`.